src/Periphery/LidoWrapper.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { WithdrawablePeriphery } from "../Helpers/WithdrawablePeriphery.sol";
+
+/// @title IStETH
+/// @notice External interface for Lido's stETH contract which supports wrapping and unwrapping wstETH
+interface IStETH is IERC20 {
+    /// @notice Unwraps wstETH into stETH
+    /// @param amount The amount of wstETH to unwrap
+    function wrap(uint256 amount) external;
+
+    /// @notice Wraps stETH into wstETH
+    /// @param amount The amount of stETH to wrap
+    function unwrap(uint256 amount) external;
+
+    /// @notice Wraps stETH into wstETH using shares
+    /// @param amount The number of shares to unwrap
+    function unwrapShares(uint256 amount) external;
+}
+
+/// @title LidoWrapper
+/// @author LI.FI (https://li.fi)
+/// @notice Wraps and unwraps Lido’s wstETH and stETH tokens
+/// @dev Be aware that Lido's L2 `wrap`/`unwrap` naming is reversed from the typical expectation.
+/// @custom:version 1.0.0
+contract LidoWrapper is WithdrawablePeriphery {
+    /// @notice Reference to the L2 stETH contract
+    IStETH public immutable ST_ETH;
+
+    /// @notice Address of the wstETH token contract
+    address public immutable WST_ETH_ADDRESS;
+
+    error ContractNotYetReadyForMainnet();
+
+    /// @notice Constructor
+    /// @param _stETHAddress The address of the stETH token on L2
+    /// @param _wstETHAddress The address of the bridged wstETH token on L2
+    /// @param _owner The address of the contract owner
+    constructor(
+        address _stETHAddress,
+        address _wstETHAddress,
+        address _owner
+    ) WithdrawablePeriphery(_owner) {
+        ST_ETH = IStETH(_stETHAddress);
+        WST_ETH_ADDRESS = _wstETHAddress;
+
+        // the wrap/unwrap functions are different on mainnet
+        if (block.chainid == 1) revert ContractNotYetReadyForMainnet();
+
+        // Approve stETH contract to pull wstETH from this contract
+        IERC20(WST_ETH_ADDRESS).approve(address(ST_ETH), type(uint256).max);
+    }
+
+    /// @notice Wraps stETH into wstETH
+    /// @dev Transfers `_amount` stETH from caller, unwraps it via the stETH contract (which yields wstETH), and returns wstETH to the caller.
+    /// @param _amount The amount of stETH to wrap into wstETH
+    function wrapStETHToWstETH(uint256 _amount) external {
+        // Pull stETH from sender
+        IERC20(address(ST_ETH)).transferFrom(
+            msg.sender,
+            address(this),
+            _amount
+        );
+
+        // Call `unwrap` on stETH contract to get wstETH (naming is inverted)
+        ST_ETH.unwrap(_amount);
```

Comment:
Should we check the actually received stETH amount here to avoid the rounding error issues of steth?

Reply:
we could just send the full balance to ST_ETH contract (instead of checking and expecting a certain amount).

WDYT?